### ISSUE TITLE
Constraints on weights for custom layer

### ISSUE THREAD

**Issue by NOEPG** [keras-team/keras#13112]
Hello,

I would the weights of my custom layer to be non-negative.
Unfortunately https://keras.io/constraints/ cannot be used on custom layer.

Is there a way to do this cleanly or should I include a term in my loss function that penalizes negative weights ?

Here is my custom layer.

class VectorCombination(Layer):

    def __init__(self, output_dim, **kwargs):
        self.output_dim = output_dim
        super(VectorCombination, self).__init__(**kwargs)

    def build(self, input_shape):
        self.kernel = self.add_weight(name='kernel', shape=(input_shape[2],), initializer='uniform', trainable=True)
        super(VectorCombination, self).build(input_shape)

    def call(self, x):
        o = [K.squeeze(K.dot(x, K.expand_dims(self.kernel)), axis=-1)]
        return o

    def compute_output_shape(self, input_shape):
        return (input_shape[0], self.output_dim)

This is simply a linear combination of very large vectors.
This layer is followed by a tanh activation.
The idea is to combine several waveforms.

Thank You for your help.
